### DISCUSSION TITLE
Can't find a codec for CodecCacheKey{clazz=class domain.Product, types=null}.

### DISCUSSION THREAD

**Discussion by sh1bz** [micronaut-projects/micronaut-core#9661] [Q&A]
MongoDB 
Micronaut 4

I'm getting this CodefCacheKey error when Test Mode. It worked on 3.9 version i'm currently running v4.     
However if I run testFunctionRequestHandler it works and when I run testApi I get the error. 
This being said, it also works on AWS lambda native but doesn't work on AWS shadowJar.

Please help.

@Inject
    lateinit var catalogController: Controller

    @Test
    fun testFunctionRequestHandler() {
        val product = catalogController.getProduct("64afc8ea21c2f173458c93a9")
        assertNotNull(product)
    }

    @Test
    fun testApi() {
        val handler = ApiGatewayProxyRequestEventFunction()
        val request = APIGatewayProxyRequestEvent()
        request.httpMethod = "GET"
        request.path = "/catalog/product/64afc8ea21c2f173458c93a9"
        val response = handler.handleRequest(request, lambdaContext)
        assertEquals(200, response.statusCode)
        handler.applicationContext.close()
    }

**Comment by prashant-mns** [comment]
If you put your entities outside of the application package, you need to specify it in the mongo configuration:

```
mongodb:
  package-names:
    xyz
```
